src/flashbar/__tests__/utils.test.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { getItemSelection, getItemType } from '../utils';
+import { FlashbarProps, FlashType } from '../interfaces';
+
+describe('getItemSelection', () => {
+  const cases: { description: string; input: FlashType[]; expectedOutput: FlashType[] }[] = [
+    {
+      description: "All items of type 'success'",
+      input: ['success', 'success', 'success'],
+      expectedOutput: ['success', 'success', 'success'],
+    },
+    {
+      description: "All items of type 'info'",
+      input: ['info', 'info', 'info'],
+      expectedOutput: ['info', 'info', 'info'],
+    },
+    {
+      description: "All items of type 'warning'",
+      input: ['warning', 'warning', 'warning'],
+      expectedOutput: ['warning', 'warning', 'warning'],
+    },
+    {
+      description: "All items of type 'error'",
+      input: ['error', 'error', 'error'],
+      expectedOutput: ['error', 'error', 'error'],
+    },
+    {
+      description: '1 item of each color',
+      input: ['success', 'warning', 'error'],
+      expectedOutput: ['success', 'warning', 'error'],
+    },
+    {
+      description: '1 item of different type after the length limit',
+      input: ['success', 'success', 'success', 'error'],
+      expectedOutput: ['success', 'success', 'error'],
+    },
+    {
+      description: '2 items of 2 different types after the length limit',
+      input: ['success', 'success', 'success', 'error', 'info'],
+      expectedOutput: ['success', 'error', 'info'],
+    },
+    {
+      description: '1 item of one type and 2 of another type',
+      input: ['success', 'error', 'error'],
+      expectedOutput: ['success', 'error', 'error'],
+    },
+    {
+      description: '1 item of one type and 3 of another type',
+      input: ['success', 'error', 'error', 'error'],
+      expectedOutput: ['success', 'error', 'error'],
+    },
+    {
+      description: '2 items of different type but same color, and 2 items of different type and color',
+      input: ['info', 'progress', 'success', 'error'],
+      expectedOutput: ['info', 'success', 'error'],
+    },
+    {
+      description: '3 items of different color followed by more items',
+      input: ['success', 'info', 'error', 'info', 'warning', 'progress'],
+      expectedOutput: ['success', 'info', 'error'],
+    },
+  ];
+
+  for (const testCase of cases) {
```

Comment:
minor comment: You can use `it.each` instead :)

Reply:
Thanks, I didn't know about it. But this doesn't seem to offer a way to let each test have a distinct description, which would make it a bit harder to debug if one or some of them fails.